FillableFieldworkAIDriver.lua

```diff
@@ -57,86 +62,117 @@ end
 
 --- Drive the refill part of the course
 function FillableFieldworkAIDriver:driveUnloadOrRefill()
+	if self:getSiloSelectedFillTypeSetting():isEmpty() then 
+		self:setSpeed(0)
+		self:setInfoText('NO_SELECTED_FILLTYPE')
+		return
+	else
+		self:clearInfoText('NO_SELECTED_FILLTYPE')
+	end
 	local isNearWaitPoint, waitPointIx = self.course:hasWaitPointWithinDistance(self.ppc:getCurrentWaypointIx(), 5)
-
-	self:searchForRefillTriggers()
+	self.waitPointIx = waitPointIx
+	local isInWaitPointRange = self.waitPointIx and self.waitPointIx+8 >self.ppc:getCurrentWaypointIx() or isNearWaitPoint 
+	if self:is_a(FieldSupplyAIDriver) then
```

Comment:
Checking for a derived class instance is not the best practice in OO, instead, this whole section, maybe from line 60 should be replaced by a function which can then be overridden in FieldSupplyAIDriver, for example have FillableFieldworkAIDriver:activateTrigger() and a FieldSupplyAIDriver:activateTrigger() for the FieldSupplyAIDriver specific behavior.

An example of something similar is the shouldReturnToFirstPoint() function.